DOC-5535 Lettuce vector set examples (#1995)
redis/docs: content/develop/clients/lettuce/vecsets.md

{{< /clients-example >}}

This returns the following list of elements (formatted slightly for clarity):

```
['Masako Natsume', 'Chaim Topol', 'Linus Pauling',
'Marie Fredriksson', 'Maryam Mirzakhani', 'Marie Curie',
'Freddie Mercury', 'Paul Erdos']
```

The first two people in the list are the two actors, as expected, but none of the
people from Linus Pauling onward was especially well-known for acting (and there certainly
isn't any information about that in the short description text).
As it stands, the search attempts to rank all the elements in the set, based
on the information contained in the embedding model.
You can use the `count` parameter of `vsim()` to limit the list of elements
to just the most relevant few items:

{{< clients-example set="home_vecsets" step="limited_query" lang_filter="Java-Async,Java-Reactive" >}}
{{< /clients-example >}}

The reason for using text embeddings rather than simple text search
is that the embeddings represent semantic information. This allows a query
to find elements with a similar meaning even if the text is
different. For example, the word "entertainer" doesn't appear in any of the

> descriptions but if you use it as a query, the actors and musicians are ranked

Reviewer: ```suggestion
descriptions, but if you use it as a query, the actors and musicians are ranked
```

highest in the results list:

{{< clients-example set="home_vecsets" step="entertainer_query" lang_filter="Java-Async,Java-Reactive" >}}
{{< /clients-example >}}

Similarly, if you use "science" as a query, you get the following results:

```
['Marie Curie', 'Linus Pauling', 'Maryam Mirzakhani',
'Paul Erdos', 'Marie Fredriksson', 'Freddie Mercury', 'Masako Natsume',
'Chaim Topol']
```

> The scientists are ranked highest but they are then followed by the
> mathematicians. This seems reasonable given the connection between mathematics
> and science.

Reviewer: ```suggestion
The scientists are ranked highest, followed by the
mathematicians. This ranking seems reasonable given the connection between mathematics and science.
```
